pydantic/_internal/_generate_schema.py

```diff
@@ -53,6 +54,31 @@
 ]
 
 
+class _CommonField(TypedDict, total=False):
```

Comment:
I'm guessing you guys aren't going to like this approach; I'm happy to rework it in whatever way makes sense, I just want to make sure that we can share as much of the field-building logic as possible between typed dict / model fields and dataclass fields

Reply:
When I first saw this, I didn't like it. But seeing the actual usage I'm more okay with it. I think what would really help is a docstring / comment giving some background and explaining why it's beneficial to have this class/function.

Reply:
i feel vaguely the same as @adriangb, but can we move it to the bottom (or near the bottom) of the file, rather than up here.